Generate Serialization for ObjectIdentifierReference and ObjectIdentifierWriteReference

NKRosario opened the pull request:
#### 641bc812809ad7b2cfb3ee166fb16753e53d4ffe
<pre>
Generate Serialization for ObjectIdentifierReference and ObjectIdentifierWriteReference
<a href="https://bugs.webkit.org/show_bug.cgi?id=268322">https://bugs.webkit.org/show_bug.cgi?id=268322</a>
<a href="https://rdar.apple.com/121877374">rdar://121877374</a>

Reviewed by Alex Christensen.

* Source/WebKit/CMakeLists.txt:
* Source/WebKit/DerivedSources-input.xcfilelist:
* Source/WebKit/DerivedSources.make:
* Source/WebKit/Platform/IPC/ObjectIdentifierReference.serialization.in: Added.
* Source/WebKit/Platform/IPC/ObjectIdentifierReferenceTracker.h:
(IPC::ObjectIdentifierReference::isHashTableDeletedValue const):
(IPC::ObjectIdentifierReference::encode const): Deleted.
(IPC::ObjectIdentifierReference::decode): Deleted.
(IPC::ObjectIdentifierWriteReference::encode const): Deleted.
(IPC::ObjectIdentifierWriteReference::decode): Deleted.
* Source/WebKit/Scripts/generate-serializers.py:
(get_alias_namespace):
(generate_forward_declarations):
(parse_serialized_types):
(main):
* Source/WebKit/Shared/WTFArgumentCoders.serialization.in:
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Tools/TestWebKitAPI/Tests/IPC/ThreadSafeObjectHeapTests.cpp:

Canonical link: <a href="https://commits.webkit.org/273849@main">https://commits.webkit.org/273849@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/50bf11343d0d07c3e381b442b6d6eeeb4b3f4918

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/36420 "5 style errors") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/15367 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/38636 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/39121 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/32713 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/37648 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/18133 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/12401 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/31358 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/36979 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/13192 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/32269 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/11363 "Passed tests") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/36553 "Passed tests") | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/11556 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/32506 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/40366 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/33256 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/32851 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/37323 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/11608 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/9600 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/35434 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/13315 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/8353 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/12057 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/12476 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

achristensen07 commented on Source/WebKit/WebKit.xcodeproj/project.pbxproj:
```diff
@@ -3,7 +3,7 @@
 	classes = {
 	};
-	objectVersion = 54;
+	objectVersion = 55;
```
This change should probably be reverted.

achristensen07 commented on Source/WebKit/Platform/IPC/ObjectIdentifierReferenceTracker.h:
```diff
@@ -60,22 +61,8 @@ class ObjectIdentifierReference {
-        return ObjectIdentifierReference { WTFMove(*identifier), *version };
-    }
 private:
+    friend struct IPC::ArgumentCoder<ObjectIdentifierReference, void>;
```
This is probably not necessary because there are accessors for the identifier and version.

achristensen07 commented on Source/WebKit/Platform/IPC/ObjectIdentifierReferenceTracker.h:
```diff
@@ -125,24 +112,8 @@ class ObjectIdentifierWriteReference {
-    }
-
 private:
+    friend struct IPC::ArgumentCoder<ObjectIdentifierWriteReference, void>;
```
Ditto.  Use public accessors.

achristensen07 reviewed (commented):
Could you add to Source/WebKit/Scripts/webkit/tests/TestSerializedType.serialization.in then run "make -C Source/WebKit/Scripts/webkit/tests" to show what the changes to the python do to the generated C++?

NKRosario committed:
Generate Serialization for ObjectIdentifierReference and ObjectIdentifierWriteReference
https://bugs.webkit.org/show_bug.cgi?id=268322
rdar://121877374

Reviewed by Alex Christensen.

* Source/WebKit/CMakeLists.txt:
* Source/WebKit/DerivedSources-input.xcfilelist:
* Source/WebKit/DerivedSources.make:
* Source/WebKit/Platform/IPC/ObjectIdentifierReference.serialization.in: Added.
* Source/WebKit/Platform/IPC/ObjectIdentifierReferenceTracker.h:
(IPC::ObjectIdentifierReference::isHashTableDeletedValue const):
(IPC::ObjectIdentifierReference::encode const): Deleted.
(IPC::ObjectIdentifierReference::decode): Deleted.
(IPC::ObjectIdentifierWriteReference::encode const): Deleted.
(IPC::ObjectIdentifierWriteReference::decode): Deleted.
* Source/WebKit/Scripts/generate-serializers.py:
(get_alias_namespace):
(generate_forward_declarations):
(parse_serialized_types):
(main):
* Source/WebKit/Shared/WTFArgumentCoders.serialization.in:
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Tools/TestWebKitAPI/Tests/IPC/ThreadSafeObjectHeapTests.cpp:

Canonical link: https://commits.webkit.org/273849@main
